### PR TITLE
#62: instrument retry policy

### PR DESCRIPTION
**Pull request by Cokemonkey11**

### CODE REVIEW

File: src/main/java/net/moonlightflower/wc3libs/bin/GameExe.java [hunk 1]
```diff
@@ -12,17 +15,37 @@
 import java.io.*;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
+import java.time.temporal.ChronoUnit;
 import java.util.Arrays;
-import java.util.Collections;
```
Comment (Cokemonkey11) [inline src/main/java/net/moonlightflower/wc3libs/bin/GameExe.java:16]: Drive-by fix: unused imports

File: src/main/java/net/moonlightflower/wc3libs/bin/GameExe.java [hunk 2]
```diff
@@ -31,59 +54,66 @@ public static String getVersionString(@Nonnull File file) throws IOException {
...
+                " datafile"+
+                " where"+
+                String.format(" name=\"%s\"", file.getAbsolutePath().replaceAll("\\\\", "\\\\\\\\"))+
+                " get"+
+                " Version"+
+                " /value"+
+                String.format(" 1>\"%s\"", tmpOut.getAbsolutePath().replaceAll("\\\\", "\\\\\\\\"));
 
-            Files.write(tmpBat.toPath(), Arrays.asList("chcp 65001", query, "EXIT /B %ERRORLEVEL%"));
+        //System.out.println("load " + query);
```
Comment (Cokemonkey11) [inline src/main/java/net/moonlightflower/wc3libs/bin/GameExe.java:80]: Drive-by fix: commented out code

File: src/main/java/net/moonlightflower/wc3libs/bin/GameExe.java [hunk 3]
```diff
@@ -31,59 +54,66 @@ public static String getVersionString(@Nonnull File file) throws IOException {
...
+        if (proc.exitVal() != 0) {
+            throw new IOException(proc.getErrString());
+        }
 
-            BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(tmpOut), StandardCharsets.UTF_8));
+        StringBuilder sb = new StringBuilder();
+        InputStreamReader inputStream = new InputStreamReader(new FileInputStream(tmpOut), StandardCharsets.UTF_8);
 
-            StringBuilder sb = new StringBuilder();
+        try(BufferedReader reader = new BufferedReader(inputStream)) {
```
Comment (Cokemonkey11) [inline src/main/java/net/moonlightflower/wc3libs/bin/GameExe.java:95]: Drive-by fix: unclosed BufferedReader

**Comment by Frotty** [comment]
as mentioned in chat, retrying the failing exe parsing won't do us any good here.
There now exists a new version for the peparser util which I made a PR for.
